src/UI/SignTransactionsModals/index.tsx

```diff
@@ -71,39 +58,55 @@ export const SignTransactionsModals = ({
     verifyReceiverScam
   };
 
-  if (signError || hasTransactions) {
-    switch (loginMethod) {
-      case LoginMethodsEnum.ledger:
-        return CustomConfirmScreens?.Ledger ? (
-          <CustomConfirmScreens.Ledger {...signProps} />
-        ) : (
-          <SignWithLedgerModal {...signProps} />
-        );
-
-      case LoginMethodsEnum.walletconnect:
-      case LoginMethodsEnum.walletconnectv2:
-        return CustomConfirmScreens?.WalletConnect ? (
-          <CustomConfirmScreens.WalletConnect {...signProps} />
-        ) : (
-          <SignWithWalletConnectModal {...signProps} />
-        );
+  const ConfirmScreens: CustomConfirmScreensType = {
+    Ledger: CustomConfirmScreens?.Ledger ?? SignWithLedgerModal,
+    WalletConnect:
+      CustomConfirmScreens?.WalletConnect ?? SignWithWalletConnectModal,
+    Extension: CustomConfirmScreens?.Extension ?? SignWithExtensionModal,
+    Extra: CustomConfirmScreens?.Extra
+  };
 
-      case LoginMethodsEnum.extension:
-        return CustomConfirmScreens?.Extension ? (
-          <CustomConfirmScreens.Extension {...signProps} />
-        ) : (
-          <SignWithExtensionModal {...signProps} />
-        );
+  const shouldShowTransactionStatusToast =
+    Boolean(signError) || Boolean(cancelTransactionsMessage);
 
-      case LoginMethodsEnum.extra:
-        return CustomConfirmScreens?.Extra ? (
-          <CustomConfirmScreens.Extra {...signProps} />
-        ) : null;
+  const renderScreen = useCallback(
+    (Screen?: ScreenType) => {
+      return (
+        <ConfirmationScreen
+          Screen={Screen}
+          signProps={signProps}
```

Comment:
this function receives the exact same props, just for the 1st, which is the screen. Maybe we can make this into a JSX that receives the ConfirmScreen and tries to extras as much as possible from Redux, instead of getting prop drilled?

Reply:
You're right. I will refactor this part